### DISCUSSION TITLE
Issue when building with cmake tools compared to cmake in console

### DISCUSSION THREAD

**Discussion by dgg80** [microsoft/vscode-cmake-tools#2283] [Q&A] [answered]
Hi there, I'm having an issue with cmake tools.

When I use them, to build my target in VsCode, I get different results compared to building with cmake and make from the console under linux.

I build in the console with
cmake -d "unix makefiles" path_to_sources .
make target

The targets run fine
When I build in VsCode, the output in the terminal is
/usr/bin/cmake --build target_path --config Debug --target target_name --

The result however is not working correctly, and I get an "inlining failed in call to always_inline '__mm256i _mm256-LOAD_SI256(CONST _MM256I*)': target specific option mismatch

Is there a way to configure cmake tools in a way, that it uses the command I also use in the console? Or do you have any other directions for me to solve my problem?

In case that this board is not the right place to ask my question, please tell me, where I can do it instead.

Thanks a lot.

**Comment by bobbrow** [comment, marked as answer]
I'm missing some information.  When you build in the terminal, what command do you use to generate the unix makefiles?  (e.g. I'm looking for the command that includes `-G "Unix Makefiles"`).  This is where I expect your compiler to be set and it's possible that the extension is attempting to build with a different compiler.  If you let cmake figure this out for you, you'd want to select an `[Unspecified]` Kit in the status bar.  

![image](https://user-images.githubusercontent.com/12818240/145437461-d3bbda77-d350-4460-9415-44f825b449b9.png)

Otherwise, you'd want to select the Kit that matches what you're setting for `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER`.

If that's not it, you can inspect the cmake commands that we're invoking for you using these new VS Code commands:
![image](https://user-images.githubusercontent.com/12818240/145437687-22d8eca5-b9f3-4087-b146-59f25b063efc.png)

You can share the output of those commands here if you'd like us to troubleshoot with you further.

**Reply by dgg80** [reply]
Hey bobrow, thanks a lot for your answer. It's right, I used -G "Unix Makefiles". Right now, I'm not having access to the machine I was working with, but I'm coming back to you next thursday, I'm sorry for my belated reply and thanks a lot already for your help.

**Reply by dgg80** [reply]
Ok, right now, I'm not able to build it on the console as well. I don't know what's going on right now. Thanks a lot for your help.
